torch/testing/_internal/composite_compliance.py

```diff
@@ -6,9 +6,39 @@
 from torch.utils._pytree import tree_map, tree_flatten, tree_unflatten
 from functools import partial
 from torch.utils._python_dispatch import enable_python_mode
+import torch.autograd.forward_ad as fwAD
 from torch.overrides import enable_reentrant_dispatch
 import re
 
+# Since Forward AD doesn't work with `set_`
+# used for aliasing result (See Note [Alias Result]),
+# we temporarily disable it with `no_alias_result`
+alias_result = True
+
+@contextlib.contextmanager
+def no_alias_result():
+    global alias_result
+    alias_result = False
+    try:
+        yield
+    finally:
+        alias_result = True
+
+# enable dispatch before calling the func in
+# CCT's __torch_dispatch__ implementation
+# NOTE: This overrides the no_dispatch
+# while calling the func in __torch_dispatch__.
+# Refer: https://github.com/pytorch/pytorch/issues/75652
+dispatch = False
+@contextlib.contextmanager
+def disable_no_dispatch():
+    global dispatch
+    dispatch = True
+    try:
+        yield
+    finally:
+        dispatch = False
+
```

Comment:
These should probably not be global. The thing I'm worried about is if someone tries to run pytest in parallel if anything bad will happen.

We should probably have a function `generate_cct(dispatch, alias)` that returns a CCTensor subclass type that closes over `dispatch` and `alias`. We can modify generate_subclass_choices_args_kwargs to accept a Tensor subclass type (the returned CCTensor). Does that sound reasonable?

Reply:
That makes sense. Will get to it. Thanks!